instance/models/server.py

```diff
@@ -256,7 +257,59 @@ def update_status(self):
         raise NotImplementedError
 
 
-class OpenStackServer(Server):
+class OpenStackServerConfiguration(models.Model):
+    """
+    Configuration fields used by OpenStack Servers
+
+    Mutable on the instance but immutable on the Server/AppServer.
+    """
+    class Meta:
+        abstract = True
+
+    # openstack server settings
+    openstack_server_flavor = JSONField(
+        null=True, blank=True, default=None,
+        help_text='JSON openstack flavor selector, e.g. {"name": "vps-ssd-1"}.'
+                  ' Defaults to settings.OPENSTACK_SANDBOX_FLAVOR on server creation.',
+    )
+    openstack_server_base_image = JSONField(
+        null=True, blank=True, default=None,
+        help_text='JSON openstack base image selector, e.g. {"name": "ubuntu-12.04-ref-ul"}'
+                  ' Defaults to settings.OPENSTACK_SANDBOX_BASE_IMAGE on server creation.',
+    )
+    openstack_server_ssh_keyname = models.CharField(
+        max_length=256,
+        null=True, blank=True, default='',
+        help_text='SSH key name used when setting up access to the openstack project.'
+                  ' Defaults to settings.OPENSTACK_SANDBOX_SSH_KEYNAME on server creation.',
+    )
```

Comment:
We could just store in on the instance. In the appserver code, when you create a server, you can say `openstack_utils.create_server(flavor=self.instance.openstack_server_flavor)`.

And as Xavier mentions, we could retrieve the resulting value from the OpenStack API, rather than storing it on the server - unless there's some value in knowing what the requested flavor was at the time a historical appserver was provisioned.